cp-ksql-server/include/etc/confluent/docker/ksql-server.properties.template

```diff
@@ -0,0 +1,4 @@
+{% set kr_props = env_to_props('KSQL_', '') -%}
+{% for name, value in kr_props.iteritems() -%}
+{{name}}={{value}}
```

Comment:
Unfortunately KSQL uses prefixes for its configuration properties. I would suggest that the docker image be built so that environment variables which are prefixed with `KSQLCONF_` get written to the properties file.

With that, we will need to massage the environment variables so that.

1. Drop the leading `KSQLCONF_`
2. convert uppercase to lowercase
3. convert _ to .
4. Ignore `KSQL_OPTS`.

For example:
1. `KSQLCONF_BOOTSTRAP_SERVERS` -> `bootstrap.servers`
2. `KSQLCONF_KSQL_STREAMS_NUM_THREADS` -> `ksql.streams.num.threads`.
3. etc.

Reply:
Double prefix is also fine. I thought `KSQLCONF` would be less confusing: If you are reading a list of configs, and they are _all_ prefixed with `KSQLCONF` it is easy to spot, vs having some with a double `KSQL_` prefix and some not.

Reply:
cc @ybyzek .. would you have a preference from a developer experience perspective? I am totally split between the double `KSQL_` prefix and the `KSQLCONF` prefix.

Reply:
@apurvam , while I understand the intention of `KSQLCONF_` (easy to differentiate), for consistency with the rest of the components it should be just `KSQL_`.  We don't have `KAFKACONF_`, `CONNECTCONF_`, etc.  We also *do* have `SCHEMA_REGISTRY_SCHEMA_REGISTRY_INTER_INSTANCE_PROTOCOL` ;)

Reply:
Thanks @ybyzek .. I'm sold! `KSQL_KSQL_` it is :)